app/controllers/api/order_cycles_controller.rb

```diff
@@ -0,0 +1,57 @@
+module Api
+  class OrderCyclesController < BaseController
+    include EnterprisesHelper
+    respond_to :json
+
+    skip_authorization_check
+
+    def products
+      products = ProductsRenderer.new(
+        distributor,
+        order_cycle,
+        customer,
+        params.slice(:q, :page, :per_page)
+      ).products_json
+
+      render json: products
+    rescue ProductsRenderer::NoProducts
+      render status: :not_found, json: ''
+    end
+
+    def taxons
+      taxons = Spree::Taxon.joins(:products).
+        where(spree_products: { id: distributed_products(distributor, order_cycle, customer) }).
+        select('DISTINCT spree_taxons.*')
+
+      render json: ActiveModel::ArraySerializer.new(taxons, each_serializer: Api::TaxonSerializer)
+    end
+
+    def properties
+      properties = Spree::Property.joins(:products).
+        where(spree_products: { id: distributed_products(distributor, order_cycle, customer) }).
+        select('DISTINCT spree_properties.*')
+
+      render json: ActiveModel::ArraySerializer.new(
+        properties, each_serializer: Api::PropertySerializer
+      )
+    end
+
+    private
+
+    def distributor
+      Enterprise.find_by_id(params[:distributor])
+    end
+
+    def order_cycle
+      OrderCycle.find_by_id(params[:id])
+    end
+
+    def customer
+      @current_api_user.andand.customer_of(distributor) || nil
+    end
+
+    def distributed_products(distributor, order_cycle, customer)
+      OrderCycleDistributedProducts.new(distributor, order_cycle, customer).products_relation
+    end
+  end
+end
```

Comment:
It's pretty need that we start a new API controller but I'd really go the REST way following Rails conventions which will benefit us in many ways. Our codebase already suffers too much of fighting said approach and we pay the price IMO.

That means we should have 3 controllers under an `OrderCycles` routes namespace each implementing the `index` action. This would stick to the convention of `index`, `show`, `edit` and `delete` controller actions.

However, I'm happy to try this out and iterate but I'd ask you to please create an issue for it.